frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DisabledNativeCardHelpText/tests/common.unit.spec.ts

```diff
@@ -1,27 +1,65 @@
 import { screen } from "__support__/ui";
+import { createMockParameter } from "metabase-types/api/mocks";
 
 import { setup } from "./setup";
 
 describe("DashCardParameterMapper > DisabledNativeCardHelpText (OSS)", () => {
-  it("should show a help link when `show-metabase-links: true`", () => {
-    setup({ showMetabaseLinks: true });
+  it("should show a help message for native models", () => {
+    setup({
+      cardType: "model",
+    });
 
+    expect(screen.getByText(/Models are data sources/)).toBeInTheDocument();
     expect(
-      screen.getByText(
-        "A text variable in this card can only be connected to a text filter with Is operator.",
-      ),
+      screen.getByRole("link", { name: "Learn more" }),
     ).toBeInTheDocument();
-    expect(screen.getByText("Learn how")).toBeInTheDocument();
   });
 
-  it("should show a help link when `show-metabase-links: false`", () => {
-    setup({ showMetabaseLinks: false });
+  it.each([
+    {
+      parameter: createMockParameter({ type: "id" }),
+      message: /variable/,
+    },
+    {
+      parameter: createMockParameter({ type: "string/=" }),
+      message: /text variable/,
+    },
+    {
+      parameter: createMockParameter({ type: "number/!=" }),
+      message: /number variable/,
+    },
+    {
+      parameter: createMockParameter({ type: "date/all-options" }),
+      message: /date variable/,
+    },
+  ])(
+    "should show a help message for $parameter.type",
+    ({ parameter, message }) => {
+      setup({ parameter });
+      expect(screen.getByText(message)).toBeInTheDocument();
+      expect(
+        screen.getByRole("link", { name: "Learn how" }),
+      ).toBeInTheDocument();
+    },
+  );
 
-    expect(
-      screen.getByText(
-        "A text variable in this card can only be connected to a text filter with Is operator.",
-      ),
-    ).toBeInTheDocument();
-    expect(screen.getByText("Learn how")).toBeInTheDocument();
-  });
+  it.each([{ showMetabaseLinks: false }, { showMetabaseLinks: true }])(
+    "should show a parameter help link when `show-metabase-links`: %s",
+    ({ showMetabaseLinks }) => {
+      setup({ showMetabaseLinks });
+      expect(
+        screen.getByRole("link", { name: "Learn how" }),
+      ).toBeInTheDocument();
+    },
+  );
+
+  it.each([{ showMetabaseLinks: false }, { showMetabaseLinks: true }])(
+    "should show a model help link when `show-metabase-links`: %s",
+    ({ showMetabaseLinks }) => {
+      setup({ cardType: "model", showMetabaseLinks });
+      expect(
+        screen.getByRole("link", { name: "Learn more" }),
+      ).toBeInTheDocument();
+    },
+  );
```

Comment:
I'm confused by assertions in these 2 tests and in `DisabledNativeCardHelpText/tests/enterprise.unit.spec.ts`.
I thought that when `showMetabaseLinks` is `false` we don't want to show these links :thinking: 

Can you please explain?

Reply:
This is the existing test. It means that in OSS and EE without a token this setting is ignored.

Reply:
I think it'd be good to leave a comment with this information (or tweak name of the test) :pray:

Reply:
Updated test names